### DISCUSSION TITLE
SUPPORT FOR TAILWIND OR WINDI

### DISCUSSION THREAD

**Discussion by DentalXpres** [Budibase/budibase#2362] [Ideas]
Hi congratulations!!!

It would be great to have in builder form elements/components with modern builder of CSS:
frameworks like TAILWIND or WINDI.

**Comment by mjashanks** [comment]
I think the only solution for this would be to allow people to build their own components/component libraries and import them into Budibase.

This would be a nice feature, though I can't promise that we will have it anytime soon. at this stage, our preference is to develop out a deeper set of standard components for everyone to use.

Are there particular components that are missing from Budibase that you would like to see? Very soon, we will be adding "Global Theming", which will allow you to customise styles - and ensure that all your apps have the same theme.

**Reply by Triloworld** [reply]
There is few way you can choose:
- When customisation is hard, but getting everything inline on standard default and good perfomance. (Boostrap, Angular). Require that you must stick with options for years. Cannot be sliced to more independent parts. jQuery is now example that slicing is not possible and hard. GitHub go away with this solution
- When you can choose React way when every component have their own style and none coherence between. Is great for total unique approach. Any custom solution and generated based on framework. Enterprise one app yes - generating on demand big no in this solution. After adding new component there is plenty new styles, after removing I don’t need that styles too. 
- choose to have coalition that styles are only in class, resets in elements and class generate styles based on usage. This is Tailwind and bedt option for this project. Latest update add feature to in frontend on demand generate styles that are 100% used on pages and can change anything (pseudoelements, rwd, new thinks too)
This way all components can have strict rules (styles only in class) but in exchange can be maximum customization and standarisation. We use it, love it and in multi tier system it work miracles. Classes can be usage to any situation - even to abstract dom, vatiations, inheritance of styles or new ideas or new points to cover (acessibility, dark mode, maybe one place need to be different and size of file must be all time low). Cavetas is similar to React: its utility and need wiring and some clarification how to use. 

From 3 options choose 2 good and 1 bad point. I choose perfomance and customization. This is Tailwind mindset (can be easily swap by other generators as they can detect class and understand syntaxt). 

This way your only one bad point is to have more to do in implementing properly. Imagine that you can just enable one class attribute in all elements to don’t bother in future about „next way of styling” or „perfomance bootleneck”
